packages/ref-metrics-pmp/pyproject.toml

```
@@ -0,0 +1,40 @@
+[project]
+name = "cmip_ref_metrics_pmp"
+version = "0.1.5"
+description = "PMP metrics provider for the CMIP Rapid Evaluation Framework"
+readme = "README.md"
+authors = [
+    { name = "Jiwoo Lee", email = "jwlee@llnl.gov" }
+]
+requires-python = ">=3.10"
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Operating System :: OS Independent",
+    "Intended Audience :: Science/Research",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Topic :: Scientific/Engineering",
+]
+dependencies = [
+    "cmip_ref_core",
+    "pcmdi_metrics",
+]
+
+[tool.uv.sources]
+pmp = { git = "https://github.com/PCMDI/pcmdi_metrics" }
```

```suggestion
pcmdi_metrics = { git = "https://github.com/PCMDI/pcmdi_metrics" }
```

This tells uv to install the pcmdi_metrics package from github. The name has to match to the dependencies section above and doesn't have to be the same as the python package name as long as it is the same above. Note that this will use the default branch instead of a particular tag.

You might need to run the following to regenerate the lockfile and your virtual environment.
```
uv lock
uv sync
```

---

@lewisjared thank you for clarifying. I have a following question -- usual install practice for PMP has been through conda channel, which then installs pmp's dependencies. If REF installs packages from github repo, should I include all PMP's dependencies somewhere in REF as well?

---

Not for now. I'm perfectly happy to mock or comment out a required step until we are able to use the dependencies. We will solve that in the coming weeks because ESMValTool needs a similar install process. We will create a docker image that bundles all of the dependencies of the metrics packages